plugin/rewrite/name.go

```diff
@@ -140,45 +193,56 @@ func newNameRule(nextAction string, args ...string) (Rule, error) {
 		}
 	}
 
-	if len(args) > 3 && len(args) != 7 {
-		return nil, fmt.Errorf("response rewrites must consist only of a name rule with 3 arguments and an answer rule with 3 arguments")
+	if len(args) > 3 && !(len(args) == 7 || len(args) == 5) {
+		return nil, fmt.Errorf("response rewrites must consist only of a name rule with 3 arguments and an answer rule with 1 or 3 arguments")
+	}
+
+	auto := false
+	if len(args) > 3 {
+		if args[3] != "answer" {
+			return nil, fmt.Errorf("exceeded the number of arguments for a non-answer rule")
+		}
+		if len(args) == 5 {
+			if strings.ToLower(args[4]) != "auto" {
+				return nil, fmt.Errorf("response rewrites must consist only of a name rule with 3 arguments and an answer rule with 1 or 3 arguments")
+			}
+			auto = true
+		} else {
+			if matchType != RegexMatch {
+				return nil, fmt.Errorf("answer name mapping only for regexp rule")
+			}
+			if strings.ToLower(args[4]) != "name" {
+				return nil, fmt.Errorf("exceeded the number of arguments for a regex name rule")
+			}
+		}
 	}
 
 	if len(args) < 7 {
+		_ = auto
```

Comment:
why?

Reply:
sorry , an artefact from testing, I'll remove it